vars/buildKibanaDockerImage.groovy

```diff
@@ -0,0 +1,125 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+import org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper
+import hudson.model.Result
+import co.elastic.BuildException
+
+/**
+
+  Builds the Docker image for Kibana, from a branch or a pull Request.
+
+  buildKibanaDockerImage(branch: 'master')
+  buildKibanaDockerImage(branch: 'PR/12345')
+*/
+def call(Map args = [:]){
+  def branch = args?.branch?.trim() ? args.branch : 'master'
+  def uppercaseBranch = branch.toUpperCase()
+
+  def kibanaBranch = branch
+  if (uppercaseBranch.startsWith('PR/')) {
+    kibanaBranch = uppercaseBranch
+  }
+  log(level: 'INFO', text: "Kibana branch is: ${kibanaBranch}")
+
+  //buildKibana(branch: "${kibanaBranch}")
+}
+
+def buildKibana(Map args = [:]) {
+  def branch = args.branch
+  def deployName = normalize(branch)
+  def kibanaDockerTargetTag = !isEmptyString(args.targetTag) ? args.targetTag : getGitCommitSha()
+
+  // constant values
+  def dockerRegistry = 'docker.elastic.co'
+  def dockerRegistrySecret = 'secret/observability-team/ci/docker-registry/prod'
+  def dockerImageSource = "${dockerRegistry}/kibana/kibana"
+  def dockerImageTarget = "${dockerRegistry}/observability-ci/kibana"
```

Comment:
Should we allow them as constant or parameters?

Reply:
I preferred to have them as constants, to simplify the step, and not allowing pushing to another registry. But maybe, from an OSS standpoint, maybe it's better to have them as params. Thoughts?

Reply:
Addressed in 3d7c724a227043c26a37c85c174ef34fb226c661